components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -231,7 +231,7 @@ class GeckoEngineSession(
             throw IllegalStateException("Can only restore from GeckoEngineSessionState")
         }
 
-        if (state.actualState == null) {
+        if (state.actualState.isNullOrEmpty()) {
```

Comment:
@agi (or should I use @agi90? 🤔) Is this an assumption we can make? If the state is empty (which seems to only work in the first place because it extends `AbstractSequentialList<HistoryDelegate.HistoryItem>`) then we can assume we cannot restore it and instead just load the URL this tab is supposed to point to? This seems to be an implementation detail and therefore maybe fragile to assume?

Reply:
In general this sounds like a GV bug, we should never give you an empty `SessionState` (the current URI should be there at the very least). @droeh do you know?

Reply:
btw @agi is my work account

Reply:
talked with @droeh this should be fine as a workaround for now, we're looking into why this is happening in GV.

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1687523

Reply:
Can we add a comment and link to the BZ maybe for future reference.